elide-core/src/main/java/com/yahoo/elide/security/checks/ExtractedChecks.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.checks;
+
+import com.yahoo.elide.audit.InvalidSyntaxException;
+import com.yahoo.elide.core.PersistentResource;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+
+/**
+ * Extracted checks.
+ */
+@Slf4j
+public class ExtractedChecks {
+    private final Class<? extends Check>[] anyEntityChecks;
+    private final Class<? extends Check>[] allEntityChecks;
+    private final Class<? extends Check>[] anyFieldChecks;
+    private final Class<? extends Check>[] allFieldChecks;
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource, final Class<A> annotationClass) {
+        this(resource, annotationClass, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param field Field
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource,
```

Comment:
I have broken this up to the bare minimum of what we need (the general code is smaller now). Are you specifically referring to this constructor? This kind of code would likely either end up here or in that static method.